### PR TITLE
[Data][Docs] Merge Data glossary with Ray glossary

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

The Data glossary is mostly redundant with Key Concepts. Also, Ray Data is the only library with a standalone glossary. For better consistency and lower maintenance cost, this PR merges the Data glossary into the general Ray glossary.

The Data glossary used to be more valuable. But, after we removed tensor and simple datasets, the glossary didn't contain many terms. The remaining terms are:
* Irrelevant (e.g, "Datasource" is an internal abstraction) 
* Redundant (e.g., "Dataset (object)" and "block" are defined in Key Concepts)
* Obsolete (e.g, "record" will be replaced by "row" going forward)

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by amogkam — COMMENTED** [review]
this also needs to be removed from the toc?

**Review by amogkam — CHANGES_REQUESTED** [review]
I thought the plan was to merge with Key concept?

The global Ray glossary doesn't really seem like the right place for this information

**Comment by bveeramani** [comment]
> I thought the plan was to merge with Key concept?
> 
> The global Ray glossary doesn't really seem like the right place for this informatio

"Block" and "Dataset (object)" are already defined in "Key Concepts", and I didn't think "Batch format" was important enough to include in key concepts.

@angelinalg we wanted to also include the terms in the Ray glossary, right?

File: doc/source/ray-references/glossary.rst [hunk 1]
```diff
@@ -70,6 +69,34 @@ Ray Data specifically, please see the :ref:`Ray Data Glossary<datasets_glossary>
         learning framework (eg. Torch, TensorFlow), used to set up distributed
         data-parallel training for :ref:`Ray Train’s built-in trainers<train-api>`.
 
+    Batch format
+        The way batches of data are represented.
```
Comment (amogkam) [inline doc/source/ray-references/glossary.rst:73]: Now that this is in the global glossary can you clarify that this is refeerring to Ray Data in particular?

The way batches of data are represented in Ray datasets.

**Review by amogkam — APPROVED** [review]

**Review by angelinalg — APPROVED** [review]

**Pull request merged**